python/ray/data/read_api.py

```diff
@@ -2313,7 +2317,13 @@ def from_torch(
     Returns:
         A :class:`MaterializedDataset` containing the Torch dataset samples.
     """  # noqa: E501
-    return from_items(list(dataset))
+    return read_datasource(
```

Comment:
we also should clarify that by default this requires the underlying files to be accessible from every node in the cluster.

we should also provide a way for users to call `from_torch` if only the head node has access to the underlying files.

Reply:
Could you clarify what this means? When are files not accessible to certain nodes in the cluster?

Reply:
If there are files that are only the head node filesystem, then they cannot be accessed by worker nodes. The recommended approach should be to store the underlying files in either a shared filesystem or in cloud storage.

Reply:
Do you know if this is being done for any other data sources?